### PR TITLE
Rework routing on front-end

### PR DESCRIPTION
**Pull request by cl-codes**
### What this PR does:
- routes/app.ts now stores all routing for the application (e.g. path, component, exact(true/false), etc.)
- added a function in App.tsx to render the routes in routes/app

### Any issues with the current functionality:
- Need to add logic to the function in App.tsx, so that we only render routes applicable to the account type (i.e. depending if the account is a pet sitter or pet owner). **For now, the function renders all routes - to be updated once profile types are updated on the back-end**.

### CODE REVIEW

**Review by sundayezeilo — COMMENTED** [review]
Looks good!

File: client/src/routes/app.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import Dashboard from '../pages/Dashboard/Dashboard';
+import Settings from '../pages/Settings/Settings';
+import { AccountType } from '../types/AccountType';
+
+export interface appRouteType {
+  to: string;
+  component: React.ComponentType;
+  canView: Array<string>;
+  exact: boolean;
+}
```
Comment (moffatethan) [inline client/src/routes/app.ts]: 💯

File: client/src/routes/app.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export const appRoutes = [
```
Comment (moffatethan) [inline client/src/routes/app.ts]: nit: `export const appRoutes: AppRouteType[] = [ . . . ]`

File: client/src/routes/app.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import Login from '../pages/Login/Login';
+import Signup from '../pages/SignUp/SignUp';
+import Dashboard from '../pages/Dashboard/Dashboard';
+import Settings from '../pages/Settings/Settings';
+import { AccountType } from '../types/AccountType';
+
+export interface appRouteType {
```
Comment (moffatethan) [inline client/src/routes/app.ts]: Generally interfaces should be PascalCase so -> `AppRouteType`

**Review by moffatethan — CHANGES_REQUESTED** [review]
Excellent, thanks for working on this! Really close again just some brief changes (1 nitpick and one requested change).

Nit picks don't need to be addressed but it's just a recommendation.

File: client/src/routes/app.ts (hunk 3 above)
Comment (cl-codes) [reply client/src/routes/app.ts]: noted and updated - thanks!!

File: client/src/routes/app.ts (hunk 2 above)
Comment (cl-codes) [reply client/src/routes/app.ts]: updated this too - thanks!

**Review by moffatethan — APPROVED** [review]
Excellent 🚀

**Pull request merged**